### PR TITLE
Bug fixes to ColorPickerInput

### PR DESCRIPTION
**Pull request by 2metres**
**User-Facing Changes**
Input now can recieve and validate color strings + minor styling pass to bring into alignment with our UI

https://github.com/foxglove/studio/assets/924528/9746662b-53c1-4568-948f-19e429f8ea90

**Description**
Previously the input displayed in the sidebar was readonly and only clicking the swatch would let you change the color. Now the inputs and UI picker are all connected

Resolves FG-4534

### CODE REVIEW

**Comment by 2metres** [comment]
🤔  Fix one thing break another

File: packages/studio-base/src/components/SettingsTreeEditor/inputs/ColorPickerInput.tsx [hunk 1]
```diff
@@ -100,10 +107,18 @@ export function ColorPickerInput(props: ColorPickerInputProps): JSX.Element {
         disabled={disabled}
         placeholder={props.placeholder}
         size="small"
-        value={displayValue ?? ""}
         variant="filled"
+        value={`#${editedValue.replace("#", "")}`}
```
Comment (2metres) [inline packages/studio-base/src/components/SettingsTreeEditor/inputs/ColorPickerInput.tsx:111]: This is less than ideal. Could use help unwravelling the `useColorPickerControl` hook

File: packages/studio-base/src/components/SettingsTreeEditor/inputs/ColorPickerInput.tsx [hunk 2]
```diff
@@ -100,10 +107,18 @@ export function ColorPickerInput(props: ColorPickerInputProps): JSX.Element {
         disabled={disabled}
         placeholder={props.placeholder}
         size="small"
-        value={displayValue ?? ""}
         variant="filled"
+        value={editedValue ? `#${editedValue.replace("#", "")}` : editedValue}
```
Comment (2metres) [inline packages/studio-base/src/components/SettingsTreeEditor/inputs/ColorPickerInput.tsx:111]: This is less than ideal. Could use help unwravelling the useColorPickerControl hook

**Review by snosenzo — APPROVED** [review]

File: packages/studio-base/src/components/SettingsTreeEditor/inputs/ColorPickerInput.tsx (hunk 2 above)
Comment (2metres) [reply packages/studio-base/src/components/SettingsTreeEditor/inputs/ColorPickerInput.tsx:111]: So we need to return undefined when empty so using starts with doesn't work. I tried moving this into the reducer but i couldn't get it to work

Comment (bryfox) [reply packages/studio-base/src/components/SettingsTreeEditor/inputs/ColorPickerInput.tsx:111]: > we need to return undefined when empty

It looks like `editedValue` is typed as a string, so I don't understand this bit.

I see you're also now returning `displayValue` (which makes sense if the intent is to offload the hex display to the calling component); that sounds to me like the value you'd want to display as a text label?

**Pull request merged**